astropy/utils/compat/__init__.py

```diff
@@ -6,9 +6,13 @@
 
 Note that all public functions in the `astropy.utils.compat.misc` module are
 imported here for easier access.
+
+The content of this module is solely for internal use of ``astropy``,
+so use it at your own risk.
```

Comment:
maybe be more specific:

"The content of this module is solely for internal use of ``astropy``  and subject to changes without deprecations. Do not use it in external package and code."